app/controllers/users/email_confirmations_controller.rb

```diff
@@ -45,10 +45,20 @@ def process_successful_confirmation(email_address)
       store_from_select_email_flow_in_session
       if current_user
         flash[:success] = t('devise.confirmations.confirmed')
-        redirect_to account_url
+        if params[:request_id]
+          redirect_to sign_up_select_email_url
+        else
+          redirect_to account_url
+        end
       else
         flash[:success] = t('devise.confirmations.confirmed_but_sign_in')
-        redirect_to root_url
+        if params[:request_id]
+          StoreSpMetadataInSession.new(
+            session: session,
+            request_id: params[:request_id],
+          ).call
+        end
```

Comment:
Can you check if this works as expected if this code is removed? I'm wondering if it's necessary to include, since as long as we redirect to `root_url` with the `request_id`, the `root_url` (`SessionsController`) should _also_ be doing the same thing to store SP metadata in session.

Reply:
You're right! I'm not sure why I thought that was needed. I have removed that.